backend/inspirehep/logger/ext.py

```diff
@@ -78,7 +78,13 @@ def init_prometheus_flask_exporter(self, app):
 
         prefix = app.name
         metrics_flask = GunicornInternalPrometheusMetrics.for_app_factory(
-            defaults_prefix=prefix, group_by=url_rule
+            defaults_prefix=prefix,
+            group_by=url_rule,
+            static_labels={
+                "source": lambda request: "external"
+                if "X-Unique-ID" in request.headers
+                else "internal"
```

Comment:
@benjamin-bergia is it going to work like this?

Reply:
I guess but I don't really understand where is request is coming from